### PR TITLE
[Typography] added boldItalic font to optional protocol of the TypographyFontLoading protocol

### PR DESCRIPTION
**Pull request by randallli**
This connivence method lets the Typography's FontLoader be the source of truth for fonts styles.

### CODE REVIEW

**Comment by randallli** [comment]
@ianegordon i'd like to try and get this in before the release today.

File: components/Typography/src/MDCTypography.m [hunk 1]
```diff
@@ -178,6 +178,13 @@ - (UIFont *)italicFontOfSize:(CGFloat)fontSize {
   return [UIFont italicSystemFontOfSize:fontSize];
 }
 
+- (UIFont *)boldItalicFontOfSize:(CGFloat)fontSize {
+  UIFont *bold = [self regularFontOfSize:fontSize];
```
Comment (ianegordon) [inline components/Typography/src/MDCTypography.m]: This naming seems a bit surprising.  bold = regularFont?  And there is no type identifier.
Consider s/bold/regularFont/

**Review by ianegordon — CHANGES_REQUESTED** [review]
Comment inline.

Consider adding documentation that fontSize must be > 0.0

File: components/Typography/src/MDCTypography.m (hunk 1 above)
Comment (randallli) [reply components/Typography/src/MDCTypography.m]: done

**Comment by randallli** [comment]
@googlebot i am ok with these changes being contributed to google

**Review by ianegordon — APPROVED** [review]

**Pull request merged**